nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -433,21 +479,50 @@ def _setup_dataloader_from_config(self, cfg: DictConfig):
                 drop_last=cfg.get("drop_last", False),
             )
         else:
-            dataset = TranslationDataset(
-                dataset_src=str(Path(cfg.src_file_name).expanduser()),
-                dataset_tgt=str(Path(cfg.tgt_file_name).expanduser()),
-                tokens_in_batch=cfg.tokens_in_batch,
-                clean=cfg.get("clean", False),
-                max_seq_length=cfg.get("max_seq_length", 512),
-                min_seq_length=cfg.get("min_seq_length", 1),
-                max_seq_length_diff=cfg.get("max_seq_length_diff", 512),
-                max_seq_length_ratio=cfg.get("max_seq_length_ratio", 512),
-                cache_ids=cfg.get("cache_ids", False),
-                cache_data_per_node=cfg.get("cache_data_per_node", False),
-                use_cache=cfg.get("use_cache", False),
-                reverse_lang_direction=cfg.get("reverse_lang_direction", False),
-            )
-            dataset.batchify(self.encoder_tokenizer, self.decoder_tokenizer)
+            if not self.multilingual:
+                src_file_list = [cfg.src_file_name]
+                tgt_file_list = [cfg.tgt_file_name]
+            else:
+                src_file_list = cfg.src_file_name
+                tgt_file_list = cfg.tgt_file_name
+
+            if len(src_file_list) != len(tgt_file_list):
+                raise ValueError(
+                    'The same number of filepaths must be passed in for source and target while training multilingual.'
+                )
+
+            datasets = []
+            for idx, src_file in enumerate(src_file_list):
+                dataset = TranslationDataset(
+                    dataset_src=str(Path(src_file).expanduser()),
+                    dataset_tgt=str(Path(tgt_file_list[idx]).expanduser()),
+                    tokens_in_batch=cfg.tokens_in_batch,
+                    clean=cfg.get("clean", False),
+                    max_seq_length=cfg.get("max_seq_length", 512),
+                    min_seq_length=cfg.get("min_seq_length", 1),
+                    max_seq_length_diff=cfg.get("max_seq_length_diff", 512),
+                    max_seq_length_ratio=cfg.get("max_seq_length_ratio", 512),
+                    cache_ids=cfg.get("cache_ids", False),
+                    cache_data_per_node=cfg.get("cache_data_per_node", False),
+                    use_cache=cfg.get("use_cache", False),
+                    reverse_lang_direction=cfg.get("reverse_lang_direction", False),
+                    prepend_id=self.multilingual_ids[idx],
+                )
+                dataset.batchify(self.encoder_tokenizer, self.decoder_tokenizer)
+                datasets.append(dataset)
+
+            if self.multilingual:
+                dataset = ConcatTranslationDataset(datasets, cfg.shuffle)
```

Comment:
Same here to specify other args to the constructor.

Reply:
Added!